import/src/test/java/com/google/cloud/healthcare/imaging/dicomadapter/StorageCommitmentServiceTest.java

```diff
@@ -0,0 +1,247 @@
+  private Association associate(
+      String serverHostname, int serverPort, String sopClass, String syntax) throws Exception {
+    AAssociateRQ rq = new AAssociateRQ();
+    rq.addPresentationContext(new PresentationContext(1, sopClass, syntax));
+    rq.setCalledAET(serverAET);
+    Connection remoteConn = new Connection();
+    remoteConn.setHostname(serverHostname);
+    remoteConn.setPort(serverPort);
+    return clientAE.connect(remoteConn, rq);
+  }
+
+  // Creates a DICOM service and returns the port it is listening on.
+  private int createDicomServer(IDicomWebClient dicomWebClient, int scuPort) throws Exception {
+    int serverPort = PortUtil.getFreePort();
+
+    DicomServiceRegistry serviceRegistry = new DicomServiceRegistry();
+    serviceRegistry.addDicomService(new BasicCEchoSCP());
+    AetDictionary aetDict = new AetDictionary(new AetDictionary.Aet[]{
+        new AetDictionary.Aet(clientAET, clientHostname, scuPort)});
+    StorageCommitmentService cmtService = new StorageCommitmentService(dicomWebClient, aetDict);
+    serviceRegistry.addDicomService(cmtService);
+
+    Device serverDevice = DeviceUtil.createServerDevice(serverAET, serverPort, serviceRegistry);
+    serverDevice.bindConnections();
+    return serverPort;
+  }
+
+  // Creates a SCU service and returns the port it is listening on.
+  private Device createSCUDevice(int listenerPort, Attributes expectReportAttrs,
+      CompletableFuture<Boolean> checkFuture) throws Exception {
+    DicomServiceRegistry serviceRegistry = new DicomServiceRegistry();
+    serviceRegistry.addDicomService(new BasicCEchoSCP());
+    StorageCommitmentSCUService scuService = new StorageCommitmentSCUService(expectReportAttrs,
+        checkFuture);
+    serviceRegistry.addDicomService(scuService);
+
+    Device serverDevice = DeviceUtil.createServerDevice(clientAET, listenerPort, serviceRegistry,
+        new TransferCapability(null,
+            UID.StorageCommitmentPushModelSOPClass,
+            TransferCapability.Role.SCU,
+            UID.ExplicitVRLittleEndian));
+    serverDevice.bindConnections();
+    return serverDevice;
+  }
+
+  private class StorageCommitmentSCUService extends AbstractDicomService {
+
+    private final Attributes expectReportAttrs;
+    private final CompletableFuture<Boolean> checkFuture;
+
+    public StorageCommitmentSCUService(Attributes expectReportAttrs,
+        CompletableFuture<Boolean> checkFuture) {
+      super(UID.StorageCommitmentPushModelSOPClass);
+      this.expectReportAttrs = expectReportAttrs;
+      this.checkFuture = checkFuture;
+    }
+
+    @Override
+    protected void onDimseRQ(Association as, PresentationContext pc, Dimse dimse, Attributes cmd,
+        Attributes data) throws IOException {
+
+      if (dimse != Dimse.N_EVENT_REPORT_RQ) {
+        throw new DicomServiceException(Status.UnrecognizedOperation);
+      }
+
+      int eventTypeID = cmd.getInt(Tag.EventTypeID, 0);
+      if (eventTypeID != 1 && eventTypeID != 2) {
```

Comment:
Can we check if event type id is 1 for success and 2 for a failure?

Reply:
Ok